packages/typography/src/Link/Link.tsx

```diff
@@ -21,15 +26,23 @@ import {
 } from './Link.styles';
 import { ArrowAppearance, LinkProps } from './Link.types';
 
-const Link = Polymorphic<LinkProps>(
+type LinkRenderProps = PolymorphicPropsWithRef<'span', LinkProps>;
+
+type AnchorLikeProps = PolymorphicProps<'a', LinkProps>;
+
+const hasAnchorLikeProps = (
+  props: LinkRenderProps | AnchorLikeProps,
+): props is AnchorLikeProps => {
+  return (props as AnchorLikeProps).href !== undefined;
+};
```

Comment:
Should we export `AnchorLike` from `Polymorphic` instead?

Reply:
We could if you think it'll be helpful in other places?